language/vm/vm-runtime/src/block_processor.rs

```diff
@@ -60,6 +50,18 @@ pub fn execute_user_transaction_block<'alloc>(
     let mut data_cache = BlockDataCache::new(data_view);
     let mut result = vec![];
 
+    // If we fail to load the gas schedule, then we fail to process the block.
```

Comment:
I am usually not too keen on the comment I am about to make (in fact **deeply** and **profoundly** dislike them) but this match is a bit difficult to read for me (TIL  you can use multiple identical cases in a `match` with an `if`). Would this be more clear? (assuming there are no other reason that wouldn't make this work, e.g. closing over the `data_view`)
```
let gas_schedule = load_gas_schedule(&module_cache, &data_cache).or_else(|_| {
    if data_view.is_genesis() {
        Ok(CostTable::zero())
    } else {
        Err(VMStatus::new(StatusCode::VM_STARTUP_FAILURE)
            .with_sub_status(sub_status::VSF_GAS_SCHEDULE_NOT_FOUND))
    }
})?;

```
anyway make it your call

Reply:
I think this would work.

At least to me, the multiple matches make the behavior of this code clearer than the `or_else` with an `if` inside of it. However.... If we ask the question "after we fix writesets, how do we want this code to look" I think the Rust-y way of doing it would be using `or_else`. I'll update to use that (and thanks for the suggestion!). (and assuming it all works, which it should).